mycroft/tts/cache.py

```diff
@@ -156,6 +162,16 @@ def __init__(self, tts_config, tts_name, audio_file_type):
             str(self.temporary_cache_dir), permissions=0o755
         )
 
+    def __contains__(self, sha):
+        """The cache contains a SHA if it knows of it and it exists on disk."""
+        if sha not in self.cached_sentences:
+            return False  # Doesn't know of it
+        else:
+            # Audio file must exist, phonemes are optional.
+            audio, phonemes = self.cached_sentences[sha]
+            return (audio.exists() and
+                    phonemes is None or phonemes.exists())
```

Comment:
The tests made me realise that this isn't quite right. I think we need parentheses around the 2nd line
eg
`A and (B or C)`
Otherwise it gets evaluated as 
`(A and B) or C`

eg
```
>>> False and True or True
True
```

Reply:
Ah, quite right. will fix